### ISSUE TITLE
Spring @WebMvcTest stopped working with ShedLock 6.0.1

### ISSUE THREAD

**Issue by waschmittel** [lukas-krecan/ShedLock#2272]
Since version 6.0.1, Spring @WebMvcTest stopped working with the following error message

```
No qualifying bean of type 'net.javacrumbs.shedlock.core.LockProvider' available: No LockProvider bean found.
```

According to documentation, @WebMvcTest `only enables auto-configuration that is relevant to MVC tests`, which is why our LockProvider Bean cannot be found.

However, it should not be needed in the first place (as it was in 5.x.x), and providing it via `@Import(SchedlockConfig.class)` is no viable solution because that requires a DataSource bean which is exactly what a @WebMvcTest should not need.

our configuration:

```java
@Configuration
public class SchedlockConfig {

    @Bean
    public LockProvider lockProvider(DataSource dataSource) {
        return new JdbcTemplateLockProvider(
                JdbcTemplateLockProvider.Configuration.builder()
                        .withJdbcTemplate(new JdbcTemplate(dataSource))
                        .usingDbTime()
                        .build()
        );
    }
}
```

Our application is annotated with `@EnableSchedulerLock(interceptMode = PROXY_METHOD)`

**Comment by lukas-krecan** [comment]
As a workaround you can move the `@EnableSchedulerLock` to the `SchedlockConfig`. Applying it to Application is not necessary.

The behavior indeed changed in 6.0.1. I like the new behavior better - to fail on startup if LockProvider is missing, but I prefer maintianing backward compatiblility more, so I will most likely revert it to the original behavior https://github.com/lukas-krecan/ShedLock/pull/2274. 

Thanks again for reporting

**Comment by lukas-krecan** [comment]
Released in 6.0.2
